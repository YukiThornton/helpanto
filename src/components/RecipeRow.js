import React from 'react';
import PropTypes from 'prop-types';
import '../styles/App.css';

const RecipeRow = ({title, selected, onClick}) => (
  <li
    onClick={onClick}
    className="Clickable"
    style={{
      color: selected ? '#0097A7' : 'gray'
    }}
  >
    {title}
  </li>
);

RecipeRow.propTypes = {
  title: PropTypes.string.isRequired,
  selected: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default RecipeRow;
