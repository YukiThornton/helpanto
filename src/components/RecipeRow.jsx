import React from 'react';
import PropTypes from 'prop-types';
import { ListItem } from 'material-ui/List';
import IconButton from 'material-ui/IconButton';
import DeleteIcon from 'material-ui/svg-icons/action/delete';
import {
  BLUE_GRAY_100,
  BLUE_GRAY_300,
  BLUE_GRAY_400,
  BLUE_GRAY_700,
} from '../constants/colors';

const itemStyle = {
  color: BLUE_GRAY_400,
  fontSize: 30,
};
const selectedItemStyle = {
  color: BLUE_GRAY_700,
  fontSize: 30,
  backgroundColor: BLUE_GRAY_100,
};
const RecipeRow = ({
  id,
  title,
  selected,
  onClick,
  onClickDeleteBtn,
}) => {
  const rightIconButton = (
    <IconButton>
      <DeleteIcon
        color={BLUE_GRAY_300}
        onClick={() => onClickDeleteBtn(id)}
      />
    </IconButton>
  );

  return (
    <ListItem
      primaryText={title}
      onClick={onClick}
      style={selected ? selectedItemStyle : itemStyle}
      rightIconButton={rightIconButton}
    />
  );
};

RecipeRow.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  selected: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
  onClickDeleteBtn: PropTypes.func.isRequired,
};

export default RecipeRow;
