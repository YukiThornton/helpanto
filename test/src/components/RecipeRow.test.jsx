import React from 'react';
import { shallow } from 'enzyme';
import RecipeRow from '../../../src/components/RecipeRow';

describe('test RecipeRow', () => {
  it('should render correctly when selected === true', () => {
    const component = (
      <RecipeRow
        id="3"
        title="Recipe title"
        selected
        onClick={() => {}}
        onClickDeleteBtn={() => {}}
      />
    );
    expect(shallow(component)).toMatchSnapshot();
  });

  it('should render correctly when selected === false', () => {
    const component = (
      <RecipeRow
        id="3"
        title="Recipe title"
        selected={false}
        onClick={() => {}}
        onClickDeleteBtn={() => {}}
      />
    );
    expect(shallow(component)).toMatchSnapshot();
  });
});
