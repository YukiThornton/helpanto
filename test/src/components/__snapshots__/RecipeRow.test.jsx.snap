// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test RecipeRow should render correctly when selected === false 1`] = `
<ListItem
  autoGenerateNestedIndicator={true}
  containerElement="span"
  disableKeyboardFocus={false}
  disabled={false}
  initiallyOpen={false}
  insetChildren={false}
  nestedItems={Array []}
  nestedLevel={0}
  onClick={[Function]}
  onKeyboardFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onNestedListToggle={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  open={null}
  primaryText="Recipe title"
  primaryTogglesNestedList={false}
  rightIconButton={
    <IconButton
      disableTouchRipple={false}
      disabled={false}
      iconStyle={Object {}}
      tooltipPosition="bottom-center"
      touch={false}
    >
      <ActionDelete
        color="#90A4AE"
        onClick={[Function]}
      />
    </IconButton>
  }
  secondaryTextLines={1}
  style={
    Object {
      "color": "#78909C",
      "fontSize": 30,
    }
  }
/>
`;

exports[`test RecipeRow should render correctly when selected === true 1`] = `
<ListItem
  autoGenerateNestedIndicator={true}
  containerElement="span"
  disableKeyboardFocus={false}
  disabled={false}
  initiallyOpen={false}
  insetChildren={false}
  nestedItems={Array []}
  nestedLevel={0}
  onClick={[Function]}
  onKeyboardFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onNestedListToggle={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  open={null}
  primaryText="Recipe title"
  primaryTogglesNestedList={false}
  rightIconButton={
    <IconButton
      disableTouchRipple={false}
      disabled={false}
      iconStyle={Object {}}
      tooltipPosition="bottom-center"
      touch={false}
    >
      <ActionDelete
        color="#90A4AE"
        onClick={[Function]}
      />
    </IconButton>
  }
  secondaryTextLines={1}
  style={
    Object {
      "backgroundColor": "#CFD8DC",
      "color": "#455A64",
      "fontSize": 30,
    }
  }
/>
`;
